adders/drizzle/config/adder.ts

```diff
@@ -176,38 +176,40 @@ export const adder = defineAdderConfig({
         {
             name: ({ typescript }) => `drizzle.config.${typescript.installed ? "ts" : "js"}`,
             contentType: "script",
-            content: ({ options, ast, common, exports, typescript, imports }) => {
+            content: ({ options, ast, common, exports, typescript, imports, object }) => {
                 imports.addNamed(ast, "drizzle-kit", { defineConfig: "defineConfig" });
 
-                const isBetterSqlite = options.sqlite === "better-sqlite3";
-                if (isBetterSqlite) {
-                    imports.addNamed(ast, "node:url", { pathToFileURL: "pathToFileURL" });
-                }
-
                 const envCheckStatement = common.statementFromString(
                     `if (!process.env.DATABASE_URL) throw new Error('DATABASE_URL is not set');`,
                 );
                 common.addStatement(ast, envCheckStatement);
 
-                // specifies the turso driver for the config
-                const driver = options.sqlite === "turso" ? "driver: 'turso'," : "";
-                const authToken = options.sqlite === "turso" ? "authToken: process.env.DATABASE_AUTH_TOKEN" : "";
-
-                const defaultExport = common.expressionFromString(`
-                    defineConfig({
-                        schema: './src/lib/server/db/schema.${typescript.installed ? "ts" : "js"}',
-                        dialect: '${options.database}', 
-                        ${driver}
-                        dbCredentials: {
-                            url: process.env.DATABASE_URL,
-                            ${authToken}
-                        },
-                        verbose: true,
-                        strict: true
-                    })
-                `);
-
-                exports.defaultExport(ast, defaultExport);
+                const fallback = common.expressionFromString("defineConfig({})");
+                const { value: exportDefault } = exports.defaultExport(ast, fallback);
+                if (exportDefault.type !== "CallExpression") return;
+
+                const objExpression = exportDefault.arguments?.[0];
+                if (!objExpression || objExpression.type !== "ObjectExpression") return;
+
+                const schemaPath = common.createLiteral(`./src/lib/server/db/schema.${typescript.installed ? "ts" : "js"}`);
+                object.overrideProperty(objExpression, "schema", schemaPath);
```

Comment:
this could probably be simplified a lot if there were a method for overriding multiple properties at once:
```
object.overrideProperties(objExpression, {
    "schema": schemaPath,
    "dialect": common.createLiteral(options.database),
    "verbose": { type: "BooleanLiteral", value: true },
    "strict": { type: "BooleanLiteral", value: true },
    "dbCredentials": object.create({
        "url": common.expressionFromString("process.env.DATABASE_URL"),
        options.sqlite === "turso" ? ...{
            "driver": common.createLiteral("turso"),
            "authToken": common.expressionFromString("process.env.DATABASE_AUTH_TOKEN")
        }: ...{}
    })
})
```

Reply:
I wonder if we should really be overriding though? If they've already got a file there, I'm not sure they'd want us to mess with it. It seems simpler and safer to just create a single string and write it out if the file doesn't exist already

Reply:
hmm, perhaps we could just override the important parts then? those would be: `dialect` and `driver`. everything else could remain untouched

Reply:
> I wonder if we should really be overriding though?

i guess it depends on the ethos of `svelte-add`. are we a "run us once and you're done" app, or can it be ran again for it to be repaired/updated/"add more stuff" (in the case of tailwind plugins)

Reply:
> hmm, perhaps we could just override the important parts then? those would be: dialect and driver. everything else could remain untouched

what happens if you have a postgres dialect and driver, but sqlite connection string and schema?

> i do love the idea of this utility though. will implement it shortly

thinking about it more, I wonder if you could do something even easier to use like this:
```
parse(`{
    "schema": ${JSON.stringify(schemaPath)},
    "dialect": ${JSON.stringify(options.database)},
    "verbose": true,
    "strict": true,
    "dbCredentials": {
        "url": common.expressionFromString("process.env.DATABASE_URL"),
        ${options.sqlite === "turso" ? 
            `"driver": "turso",
            "authToken": process.env.DATABASE_AUTH_TOKEN` : ''
        }
    }
}`)
```

> i guess it depends on the ethos of svelte-add. are we a "run us once and you're done" app, or can it be ran again for it to be repaired/updated/"add more stuff" (in the case of tailwind plugins)

we could skip updates by default and just opt-in for specific things like additional plugins where it's more adding stuff vs updating stuff

Reply:
> thinking about it more, I wonder if you could do something even easier to use like this:
> 
> ```
> parse(`{
>     "schema": ${JSON.stringify(schemaPath)},
>     "dialect": ${JSON.stringify(options.database)},
>     "verbose": true,
>     "strict": true,
>     "dbCredentials": {
>         "url": common.expressionFromString("process.env.DATABASE_URL"),
>         ${options.sqlite === "turso" ? 
>             `"driver": "turso",
>             "authToken": process.env.DATABASE_AUTH_TOKEN` : ''
>         }
>     }
> }`)
> ```

this is actually quite similar to how i originally had it before this PR 😆:

https://github.com/svelte-add/svelte-add/blob/2c75392e27bd5c47d40df16489c660cf4625b389/adders/drizzle/config/adder.ts#L196-L210

i opted to replace it because i wanted the individual properties to be a bit more dynamic and overridable rather than having it be an all or nothing change. i think what you first proposed is an excellent solution that feels very ergonomic!

Reply:
I like the now chosen approach and think it's the best fit for the future. resolving this.